src/components/ui/FileBrowser/FileList.tsx

```diff
@@ -41,25 +45,72 @@ export default function FileList({
   hideDotFiles,
   handleRightClick
 }: FileListProps): JSX.Element {
+  const { currentNavigationPath, getFileFetchPath, fetchFileAsJson } = useFileBrowserContext();
+  const { currentFileSharePath } = useZoneBrowserContext();
+
   const displayFiles = React.useMemo(() => {
     return hideDotFiles
       ? files.filter(file => !file.name.startsWith('.'))
       : files;
   }, [files, hideDotFiles]);
 
-  const hasZattrs = files.some(file => file.name === '.zattrs');
+  const [hasMultiscales, setHasMultiscales] = React.useState(false);
+  const [thumbnailSrc, setThumbnailSrc] = React.useState<string | null>(null);
+  const [neuroglancerUrl, setNeuroglancerUrl] = React.useState<string | null>(null);
+  const neuroglancerUrlPattern = "https://neuroglancer-demo.appspot.com/#!{%22layers%22:[{%22source%22:%22zarr://{URL}%22,%22name%22:%22OME-NGFF%22}]}";
+
+  React.useEffect(() => {
+    const checkZattrsForMultiscales = async () => {
+      const zattrsFile = files.find(file => file.name === '.zattrs');
+      setHasMultiscales(false);
+      if (zattrsFile && currentFileSharePath) {
+        const zattrsUrl = `${currentFileSharePath.name}/${zattrsFile.path}`;
+        const zattrsContent = await fetchFileAsJson(zattrsUrl);
+        if (zattrsContent && 'multiscales' in zattrsContent) {
+          setHasMultiscales(true);
+          const fileFetchPath = getFileFetchPath(currentNavigationPath.replace('?subpath=', '/'));
+          const imageUrl = `${window.location.origin}${fileFetchPath}`;
+          try {
+            const store = new zarr.FetchStore(imageUrl);
+            const src = await omezarr.renderThumbnail(store);
+            setThumbnailSrc(src);
+          } catch (error) {
+            console.error('Error rendering thumbnail', error);
+          }
+
+          const neuroglancerUrl = neuroglancerUrlPattern.replace("{URL}", imageUrl);
+          setNeuroglancerUrl(neuroglancerUrl);
+        }
+      }
+    };
+
+    checkZattrsForMultiscales();
+  }, [files, fetchFileAsJson, currentFileSharePath]);
 
   return (
     <div
       className={`px-2 transition-all duration-300 ${showPropertiesDrawer ? 'mr-[350px]' : ''}`}
     >
       <FileListCrumbs />
 
-      {hasZattrs && (
+      {hasMultiscales && (
```

Comment:
The `variable && <JSX>` pattern can lead to '0' being rendered on the page. I have moved to using `variable ? <JSX> : null`

Reply:
Fixed. Great tip, thanks!